holoviews/element/comparison.py

```diff
@@ -522,14 +522,17 @@ def compare_dataset(cls, el1, el2, msg='Dataset'):
             raise AssertionError("%s not of matching length, %d vs. %d."
                                  % (msg, el1.shape[0], el2.shape[0]))
         for dim, d1, d2 in dimension_data:
-            if d1.dtype != d2.dtype:
-                cls.failureException(f"{msg} {dim.pprint_label} columns have different type."
-                                     + f" First has type {d1}, and second has type {d2}.")
-            if d1.dtype.kind in 'SUOV':
-                if list(d1) == list(d2):
-                    cls.failureException(f"{msg} along dimension {dim.pprint_label} not equal.")
-            else:
-                cls.compare_arrays(d1, d2, msg)
+            try:
+                np.testing.assert_equal(d1, d2)
```

Comment:
This is mainly to avoid exceptions if one type is `int` and the other is `float`.